src/Sulu/Component/Content/Types/Block/ScheduleBlockVisitor.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Types\Block;
+
+use Sulu\Bundle\HttpCacheBundle\CacheLifetime\CacheLifetimeRequestEnhancer;
+use Sulu\Component\Content\Compat\Block\BlockPropertyType;
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+
+class ScheduleBlockVisitor implements BlockVisitorInterface
+{
+    /**
+     * @var RequestAnalyzerInterface
+     */
+    private $requestAnalyzer;
+
+    /**
+     * @var CacheLifetimeRequestEnhancer
+     */
+    private $cacheLifetimeRequestEnhancer;
+
+    public function __construct(
+        RequestAnalyzerInterface $requestAnalyzer,
+        CacheLifetimeRequestEnhancer $cacheLifetimeRequestEnhancer
+    ) {
+        $this->requestAnalyzer = $requestAnalyzer;
+        $this->cacheLifetimeRequestEnhancer = $cacheLifetimeRequestEnhancer;
+    }
+
+    public function visit(BlockPropertyType $block): ?BlockPropertyType
+    {
+        $blockPropertyTypeSettings = $block->getSettings();
+
+        if (!\is_array($blockPropertyTypeSettings)
+            || !isset($blockPropertyTypeSettings['schedules_enabled'])
+            || !$blockPropertyTypeSettings['schedules_enabled']
+            || !isset($blockPropertyTypeSettings['schedules'])
+            || !\is_array($blockPropertyTypeSettings['schedules'])
+        ) {
+            return $block;
+        }
+
+        $now = $this->requestAnalyzer->getDateTime();
+        $nowTimestamp = $now->getTimestamp();
+
+        $returnBlock = false;
+
+        foreach ($blockPropertyTypeSettings['schedules'] as $schedule) {
+            switch ($schedule['type']) {
+                case 'fixed':
+                    $start = new \DateTime($schedule['start']);
+                    $end = new \DateTime($schedule['end']);
```

Comment:
this will lead to an exception if the user does not set `start` or `end` in the admin interface. can we make them mandatory somehow?

Reply:
Actually I think we've said that you should be able to only set a start or end date... I think I am going to implement it this way.

Reply:
👍